ledger-core/virtual/integration/method_test.go

```diff
@@ -1096,35 +1098,94 @@ func Test_CallMethodWithBadIsolationFlags(t *testing.T) {
+	auth := authentication.NewService(ctx, jetCoordinatorMock)
+	server.ReplaceAuthenticationService(auth)
+
+	server.Init(ctx)
 	server.IncrementPulse(ctx)
 
+	// legitimate sender
+	jetCoordinatorMock.QueryRoleMock.Return([]reference.Global{server.GlobalCaller()}, nil)
+	jetCoordinatorMock.MeMock.Return(server.GlobalCaller())
+
 	var (
-		objectLocal  = server.RandomLocalWithPulse()
-		objectGlobal = reference.NewSelf(objectLocal)
+		objectLocal       = server.RandomLocalWithPulse()
+		objectGlobal      = reference.NewSelf(objectLocal)
+		outgoing          = server.BuildRandomOutgoingWithPulse()
+		class             = gen.UniqueGlobalRef()
+		dirtyStateRef     = server.RandomLocalWithPulse()
+		dirtyState        = reference.NewSelf(dirtyStateRef)
+		validatedStateRef = server.RandomLocalWithPulse()
+		validatedState    = reference.NewSelf(validatedStateRef)
 	)
 
-	Method_PrepareObject(ctx, server, payload.Ready, objectGlobal)
+	const (
+		validatedMem = "12345"
+		dirtyMem     = "54321"
+	)
 
-	executeDone := server.Journal.WaitAnyActivityOf(&execute.SMExecute{}, 1)
+	Method_PrepareObject(ctx, server, payload.Ready, objectGlobal)
 
 	typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
 	typedChecker.VCallResult.Set(func(res *payload.VCallResult) bool { return false })
 	typedChecker.VStateReport.Set(func(res *payload.VStateReport) bool { return false })
+	typedChecker.VStateRequest.Set(func(res *payload.VStateRequest) bool {
+		report := &payload.VStateReport{
+			Status:               payload.Ready,
+			AsOf:                 0,
+			Object:               objectGlobal,
+			LatestValidatedState: validatedState,
+			LatestDirtyState:     dirtyState,
+			ProvidedContent: &payload.VStateReport_ProvidedContentBody{
+				LatestValidatedState: &payload.ObjectState{
+					Reference: validatedStateRef,
+					Class:     class,
+					State:     []byte(validatedMem),
+				},
+				LatestDirtyState: &payload.ObjectState{
+					Reference: dirtyStateRef,
+					Class:     class,
+					State:     []byte(dirtyMem),
+				},
+			},
+		}
+		server.SendPayload(ctx, report)
+		return false
+	})
 
-	{
-		pl := payload.VCallRequest{}
-		server.SendPayloadAsFuture(ctx, &pl)
+	pl := payload.VCallRequest{
+		CallType:            payload.CTMethod,
+		CallFlags:           payload.BuildCallFlags(contract.CallIntolerable, contract.CallValidated),
+		Caller:              server.GlobalCaller(),
+		Callee:              objectGlobal,
+		CallSiteDeclaration: testwallet.GetClass(),
+		CallSiteMethod:      "Accept",
+		CallOutgoing:        outgoing,
+		Arguments:           insolar.MustSerialize([]interface{}{}),
 	}
 
-	server.IncrementPulse(ctx)
+	// now we are not legitimate sender
+	jetCoordinatorMock.QueryRoleMock.Return([]reference.Global{server.RandomGlobalWithPulse()}, nil)
+	jetCoordinatorMock.MeMock.Return(server.RandomGlobalWithPulse())
 
-	testutils.WaitSignalsTimed(t, 10*time.Second, executeDone)
+	server.SendPayloadAsFuture(ctx, &pl)
```

Comment:
I strongly recommend to add WaitIdleConveyor() before sending anything

Reply:
done